crates/compiler/parse/src/ast.rs

```diff
@@ -937,6 +949,16 @@ impl<'a, 'b> RecursiveValueDefIter<'a, 'b> {
                     expr_stack.push(&condition.value);
                     expr_stack.push(&cont.value);
                 }
+                Return(return_value, after_return) => {
+                    if let Some(after_return) = after_return {
+                        expr_stack.reserve(2);
+                        expr_stack.push(&return_value.value);
+                        expr_stack.push(&after_return.value);
+                    } else {
+                        expr_stack.reserve(1);
```

Comment:
Nit: this reserve call seems unnecessary?

Reply:
Ah, I see what you mean: reservations preempt multiple resizings of the backing array, but a reservation of 1 couldn't prevent multiple resizings as there'd only ever be a max of one.

Removed!